packages/feature-qpf/lib/views/quadpatternfragments/QuadPatternFragmentsRdfView.js

```diff
@@ -43,63 +42,80 @@ class QuadPatternFragmentsRdfView extends RdfView {
-      '"Triple/Quad Pattern Fragment of the \'' + (datasource.title || '') + '\' dataset ' +
-      'containing triples matching the pattern ' + query.patternString + '."@en');
-    metadata(fragment.pageUrl, dcTerms + 'source',   datasource.url);
+    this.validateMetadata(metadata, this._createNamedNode(fragment.url), this._createNamedNode(voID + 'subset'), this._createNamedNode(fragment.pageUrl));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(rdf + 'type'), this._createNamedNode(hydra + 'PartialCollectionView'));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(dcTerms + 'title'),
+      this._createLiteral('Linked Data Fragment of ' + (datasource.title || ''), 'en'));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(dcTerms + 'description'),
+      this._createLiteral('Triple/Quad Pattern Fragment of the \'' + (datasource.title || '') + '\' dataset ' +
+      'containing triples matching the pattern ' + query.patternString + '.', 'en'));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(dcTerms + 'source'), this._createNamedNode(datasource.url));
 
     // Total pattern matches count
     var totalCount = meta.totalCount;
-    metadata(fragment.pageUrl, hydra + 'totalItems', '"' + totalCount + '"^^' + xsd + 'integer');
-    metadata(fragment.pageUrl, voID  + 'triples',    '"' + totalCount + '"^^' + xsd + 'integer');
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(hydra + 'totalItems'), this._createLiteral(totalCount));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(voID  + 'triples'), this._createLiteral(totalCount));
 
     // Page metadata
-    metadata(fragment.pageUrl, hydra + 'itemsPerPage', '"' + query.limit + '"^^' + xsd + 'integer');
-    metadata(fragment.pageUrl, hydra + 'first', fragment.firstPageUrl);
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(hydra + 'itemsPerPage'), this._createLiteral(query.limit));
+    this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(hydra + 'first'), this._createNamedNode(fragment.firstPageUrl));
     if (query.offset)
-      metadata(fragment.pageUrl, hydra + 'previous', fragment.previousPageUrl);
+      this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(hydra + 'previous'), this._createNamedNode(fragment.previousPageUrl));
     if (totalCount >= query.limit + (query.offset || 0))
-      metadata(fragment.pageUrl, hydra + 'next', fragment.nextPageUrl);
+      this.validateMetadata(metadata, this._createNamedNode(fragment.pageUrl), this._createNamedNode(hydra + 'next'), this._createNamedNode(fragment.nextPageUrl));
+  }
+
+  _createNamedNode(term) {
```

Comment:
For code clarity, I suggest to not make use of these helper functions.
Instead, you can wrap certain metadata calls in if-statements, which will make things clearer for other developers, on which things may be omitted from metadata.